pyam/core.py

```diff
@@ -817,7 +870,8 @@ def aggregate_region(self, variable, region='World', subregions=None,
         # compute aggregate over all subregions
         subregion_df = self.filter(region=subregions)
         cols = ['region', 'variable']
-        _data = _aggregate(subregion_df.filter(variable=variable).data, cols)
+        _data = _aggregate(subregion_df.filter(variable=variable).data,
+                           cols, method=method)
 
         # add components at the `region` level, defaults to all variables one
```

Comment:
please add an `if` statement that the `components` defined only at `region` level are exclusively added when using `method in [`'sum', np.sum]`

Reply:
ok.